worksheet_A/TerminalHacking.cpp

```diff
@@ -41,6 +53,75 @@ int main()
 
 	// TODO: implement the rest of the game
 
+	// Initialise variables
+	std::string guess;
+	int maxLives = 4;
+	int remainingLives = maxLives;
+
+	// Two different ways of displaying text to console
+	std::cout << "Guess the secret word." << std::endl;
+	printf("You have %i guesses. \n\n", maxLives);
+
+	while (remainingLives > 0)
+	{
+		bool isValidGuess = false;
+
+		// Recieves player guess and checks it is in the list
+		while (!isValidGuess)
+		{
+			printf("Enter your guess: ");
+			std::cin >> guess;
+			std::cout << "your guess was: " << guess << std::endl;
+
+			// Scan through the line, converting each character to upper case.
```

Comment:
Fine, but can be done in one line using `std::transform` -- see http://stackoverflow.com/a/735215